### PR TITLE
add multiple network support and network added to link

### PR DESCRIPTION
**Pull request by 0xlinus**
Closses #265

### CODE REVIEW

File: pages/pools/index.tsx [hunk 1]
```diff
@@ -127,9 +136,13 @@ const Pools: FC = () => {
 			);
 		}
 
+		if (process.env.NODE_ENV === 'production') {
+			list = list.filter(({ network }) => network === 'mainnet' || network === 'optimism');
```
Comment (Aelin-xyz) [inline pages/pools/index.tsx:140]: we should prob create an enum for the network names that we are using for the urls and use them here and in the object we are exporting for that in case we change the names in the future

File: pages/pools/index.tsx [hunk 2]
```diff
@@ -322,9 +335,9 @@ const Pools: FC = () => {
 				/>
 				<Table
 					pageIndex={pageIndex}
-					noResultsMessage={poolsQuery.isSuccess && (data?.length ?? 0) === 0 ? 'no results' : null}
+					noResultsMessage={poolsQueryWithNetwork?.length === 0 ? 'no results' : null}
 					data={data && data.length > 0 ? data : []}
-					isLoading={poolsQuery.isLoading}
+					isLoading={!poolsQueryWithNetwork?.length}
```
Comment (Aelin-xyz) [inline pages/pools/index.tsx:340]: hmm... so we will show results as each network returns data. i guess that makes more sense than making sure all the queries have loaded. The UI might update with new data coming in but that seems ok.

Comment (0xlinus) [reply pages/pools/index.tsx:340]: nah... you're right. Will create a helper to check when all queries status !== loading

File: pages/pools/index.tsx [hunk 3]
```diff
@@ -48,8 +49,18 @@ const Pools: FC = () => {
 		Number(router.query.page ?? DEFAULT_PAGE_INDEX)
 	);
 
-	const poolsQuery = useGetPoolsQuery({ networkId: network.id });
+	const poolsQuery = useGetPoolsQuery();
+	const poolsQueryWithNetwork = useMemo(() => {
+		return poolsQuery
+			.filter((q) => !!q.data)
+			.reduce((prev, current) => {
```
Comment (0xcdb) [inline pages/pools/index.tsx:56]: I'm confused. What's the goal of `reduce` here?

**Review by 0xcdb — APPROVED** [review]

File: package.json [hunk 4]
```diff
@@ -9,7 +9,7 @@
 		"lint": "eslint . --ext ts,.tsx && tsc",
 		"format": "prettier --write \"**/*.{ts,tsx}\"",
 		"lint:fix": "eslint . --ext ts,.tsx --fix",
-		"pull-manifest": "codegen-graph-ts pull $AELIN_SUBGRAPH_URL > manifest.json",
+		"pull-manifest": "codegen-graph-ts pull https://api.thegraph.com/subgraphs/name/0xlinus/aelin-optimism > manifest.json",
```
Comment (Aelin-xyz) [inline package.json:12]: lets undo this change. I also recommend using your bash_profile or something similar instead of hardcoding for this to make it easier for everyone. we can all manage our own variables however we want

File: pages/pools/index.tsx (hunk 3 above)
Comment (Aelin-xyz) [reply pages/pools/index.tsx:56]: it is to add the network name to each row. It is ok

File: queries/pools/useGetPoolsQuery.ts [hunk 5]
```diff
@@ -1,38 +1,41 @@
...
-			contributions: true,
-			dealAddress: true,
-			hasAllowList: true,
-		},
-		{},
-		networkId ? networkId : NetworkId.Mainnet
-	);
+const useGetPoolsQuery = () => {
+	return Object.entries(nameToIdMapping).map(([networkName, networkId]) => ({
+		...useGetPoolCreateds(
```
Comment (Aelin-xyz) [inline queries/pools/useGetPoolsQuery.ts:9]: does this still give you an error about when you can call hooks?

File: pages/pools/index.tsx (hunk 3 above)
Comment (0xlinus) [reply pages/pools/index.tsx:56]: that's correct. It's to add the network name to each poolQuery result object

File: queries/pools/useGetPoolsQuery.ts (hunk 5 above)
Comment (0xlinus) [reply queries/pools/useGetPoolsQuery.ts:9]: The only problem I had here was due to the `networkId` not being pass to the other hooks in `subgraph.ts`. After that was solved, all is good

**Review by Aelin-xyz — APPROVED** [review]
